stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -42,7 +42,7 @@ size(S::SparseMatrixCSC) = (S.m, S.n)
 # underlying SparseMatrixCSC
 const SparseMatrixCSCView{Tv,Ti} =
     SubArray{Tv,2,SparseMatrixCSC{Tv,Ti},
-        Tuple{Base.Slice{Base.OneTo{Int}},I}} where {I<:AbstractUnitRange}
+        Tuple{Base.Slice{Base.OneTo{Int},true},I}} where {I<:AbstractUnitRange}
```

Comment:
This (and the accompanying change below) is worth pointing out: in nearly all cases, `Base.Slice` continues working as it used to, but here parametric invariance means that this change broke this type-alias into something that is impossible to match.

So this is indeed slightly breaking.

Reply:
> So this is indeed slightly breaking.

Policy: if we run it on all registered packages and nothing breaks, then slightly breaking is fine in the next minor release, i.e. Julia 1.1.0 (but not in Julia 1.0.1, of course).

Reply:
What would it take to run this before merging this PR? I ask because if required there is an alternative: keep `Slice` meaning what it always has and a new type (`IdentityRange`?) which is the equivalent of this PR's `Slice{T,false}`.

Reply:
Yeah, introducing a new type is probably the wiser tack here.  I had been thinking that this would be less breaking, but just introducing the new type and allowing OffsetArrays to gradually begin using it seems like a better plan.